runtime/near-test-contracts/cross-contract/Cargo.toml

```diff
@@ -0,0 +1,22 @@
+[package]
+name = "cross-contract"
+version = "0.0.0"
+authors = ["Near Inc <hello@nearprotocol.com>"]
+edition = "2018"
+
+[lib]
+crate-type = ["cdylib", "rlib"]
+
+[dependencies]
+near-sdk = "3.1.0"
```

Comment:
Hm, I am unsure about this. 

On the one hand, this contract has a dedicated Cargo.lock and is re-build on CI, so depending on an sdk would work fine. It would be reproducible, won't require real cyclic dependencies, and would be easy to use (the build process for the contract is automated). 

On the other hand, it feels like this pulls in too much stuff for the thing we actually want to test. At least, there's a problem of increased compile times. And then we'd have to update Cargo.tomls when the new sdk  version is released. 

More philosophically, here we test functionality which is independent of a particular sdk, so it feels odd  that we use a specific SDK.


I wonder, can we use existing `test-contract-rs` here instead? `write_key_value` and `read_value` methods seem like they cover `simple-state` and `multi-state`. Can we use several instances of this contract to cover cross-contract as well?

Reply:
The audience and purpose is different. test-contract-rs is unit testing contract runtime, but developer almost always use near-sdk-rs to write contracts (and reproduce given its state), so these contracts are both a test and a demonstration to contract developers that how they're going to download states and reproduce execution. For example, multi-state, it used nested UnorderedMap, a common pattern in a real world contract, and it translate to multiple trie items: every key value pair in UnorderedMap is flattened to be a trie key and value. This seems non trivial to express with plain write_key_value (and even if you do so you still need to show it works in a real contract).
As regarding concern of build time and extra dependency and odd to depend on specific sdk, I think you're right, maybe make sense to place these contracts in another repo, like contract-debug-demo or just near-sdk-rs, as a highlevel, developer facing documentation thing than i test.
For now I'll locally use these contracts to test implementing vm-standalone + rust scripts reproduce

Reply:
Aha, great points! Yeah, it seems like this then belonges elsewheer. I wonder if `near-sdk-rs` would be an appropriate place, as it is essentailly an example?

Reply:
Sounds good, I'll remake examples in near-sdk-rs